### DISCUSSION TITLE
Message to future self

### DISCUSSION THREAD

**Discussion by paneq** [RailsEventStore/rails_event_store#1159] [Ideas]
```
scheduled = CreditNoteDeliveryForInvoiceScheduled.strict(
  scheduled_at: invoice.credit_note_at,
  invoice_id:   invoice.id,
  order_id:     invoice.order_id,
)
Resque.enqueue_at(invoice.credit_note_at.utc, SendCreditNoteHandler, YAML.dump(scheduled))
event_store.publish(scheduled)
```

* https://skillsmatter.com/skillscasts/5437-answering-a-question

How can we recommend doing this pattern in RES?

**Comment by paneq** [comment]
Perhaps we could do `append_to_stream`  ?

**Comment by paneq** [comment]
We can use `ActiveJob`'s feature of course instead of `Resque.enqueue_at` like I showed from my project.

**Comment by mostlyobvious** [comment]
Related: https://www.youtube.com/watch?v=WW9xVY3S7jo

**Comment by mostlyobvious** [comment]
Related: https://blog.arkency.com/modeling-passing-time-with-events/

**Comment by mostlyobvious** [comment]
Related: https://blog.arkency.com/what-ive-learnt-at-rescon/
